### PR TITLE
🐛 Plugin Init improvements

### LINKED ISSUE #436: Plugin init command fails from time to time
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]

```sh
clusterctl-operator init -i "" --core cluster-api:capi-system:v1.6.2 -b "" -c ""
Checking that Cert Manager is installed and running.
Installing cert-manager Version="v1.13.2"
Waiting for cert-manager to be available...
Checking that CAPI Operator is installed and running.
Installing CAPI operator Version="latest"
Detected latest operator version Version="v0.9.0"
Waiting for CAPI Operator to be available...
CAPI Operator is successfully installed.
Installing provider Type="core" Name="cluster-api" Version="v1.6.2" Namespace="capi-system"
Error: cannot create core provider: cannot create provider: Internal error occurred: failed calling webhook "vcoreprovider.kb.io": failed to call webhook: Post "https://capi-operator-webhook-service.capi-operator-system.svc:443/mutate-operator-cluster-x-k8s-io-v1alpha2-coreprovider?timeout=10s": dial tcp 10.96.55.215:443: connect: connection refused
```

**What did you expect to happen:**
Plugin should successfully initialize providers

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-operator version: v0.9.0
- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

/kind bug
/area plugin

### PR DESCRIPTION
**Pull request by Fedosin**
**What this PR does / why we need it**:
This PR contains two fixes for the plugin Init operation:

- Get the latest operator version from the list of all available versions

  Now we use "latest" tag, but due to recent clusterctl changes it forces to download metadata.yaml file which is not available for CAPI operator. Because of that installation fails with exceeded context deadline. To prevent this we specify a version when we create a repo object, and then pick the latest version from the list of all available versions.

- Add a short delay before deploying providers

  We have to add a short delay here to ensure that the webhooks are successfully started.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #436

/area plugin

### CODE REVIEW

File: cmd/plugin/cmd/init.go [hunk 1]
```diff
@@ -509,6 +513,9 @@ func createGenericProvider(ctx context.Context, client ctrlclient.Client, provid
 
 	log.Info("Installing provider", "Type", provider.GetType(), "Name", name, "Version", version, "Namespace", namespace)
 
+	// We have to add a short delay here to ensure that the webhooks are successfully started.
+	time.Sleep(time.Second * 5)
```
Comment (alexander-demicev) [inline cmd/plugin/cmd/init.go:517]: are we waiting for CAPI Operator deployment to run?

**Comment by eljohnson92** [comment]
We have run into this issue as well trying to install the operator and then immediately trying to create a `coreprovider` resource. could this be solved alternatively through a podReadiness probe that waited until the webhook service was up to mark the operator pod as ready?

**Review by Danil-Grigorev — APPROVED** [review]
/lgtm

**Review by alexander-demicev — APPROVED** [review]
/approve

**Pull request merged**